improve `loadnetwork` roller ux

1. roller asks to provide a private key, instead it would be nice to have it optional, a new private key should be generated it the user doesn't want to import an existing one.

2. if an unfunded private key is provided, `roller` panics, instead it should gracefully ask to fund until the wallet is funded

```
press 'y' when the wallet are funded [y/N]: Yes
panic: WeaveVM wallet need to be fund!
```

--- a/data_layer/aptos/aptos.go
+++ b/data_layer/aptos/aptos.go
@@ -71,6 +71,9 @@ func NewAptos(root string) *Aptos {
 			panic(fmt.Errorf("DA network configuration not found for: %s", aptConfig.Network))
 		}
 
+		aptConfig.RpcEndpoint = daData.RpcUrl
+		aptConfig.Root = root
+
 		useExistingAPTWallet, _ := pterm.DefaultInteractiveConfirm.WithDefaultText(
 			"would you like to import an existing APT wallet?",
 		).Show()
@@ -99,28 +102,29 @@ func NewAptos(root string) *Aptos {
 		pterm.DefaultSection.WithIndentCharacter("🔔").Println("Please fund your APT wallet have privatekey below")
 		pterm.DefaultBasicText.Println(pterm.LightGreen(aptConfig.PrivateKey))
 
-		proceed, _ := pterm.DefaultInteractiveConfirm.WithDefaultValue(false).
-			WithDefaultText(
-				"press 'y' when the wallets are funded",
-			).Show()
+		for {
+			proceed, _ := pterm.DefaultInteractiveConfirm.WithDefaultValue(false).
+				WithDefaultText(
+					"press 'y' when the wallet are funded",
+				).Show()
 
-		if !proceed {
-			panic(fmt.Errorf("APT addr need to be fund!"))
+			if !proceed {
+				panic(fmt.Errorf("APT addr need to be fund!"))
+			}
+
+			balance, err := aptConfig.getBalance()
+			if err != nil {
+				pterm.Println("Error getting balance:", err)
+				continue
+			}
+
+			if balance > 0 {
+				pterm.Println("Wallet funded with balance:", balance)
+				break
+			}
+
+			pterm.Println("APT wallet need to be fund!")
 		}
-
-		aptConfig.RpcEndpoint = daData.RpcUrl
-		aptConfig.Root = root
-
-		balance, err := aptConfig.getBalance()
-		if err != nil {
-			panic(err)
-		}
-
-		if balance == 0 {
-			panic(fmt.Errorf("APT wallet need to be fund!"))
-		}
-
-		pterm.Println("APT Balance: ", balance)
 
 		pterm.Warning.Print("You will need to save Private Key to an environment variable named APT_PRIVATE_KEY")
 

--- a/data_layer/aptos/aptos.go
+++ b/data_layer/aptos/aptos.go
@@ -91,9 +91,6 @@ func NewAptos(root string) *Aptos {
 
 			privKeyHex := hex.EncodeToString(privKey.Seed())
 			aptConfig.PrivateKey = privKeyHex
-			if err != nil {
-				panic(err)
-			}
 
 			fmt.Printf("\t%s\n", aptConfig.PrivateKey)
 			fmt.Println()

--- a/data_layer/aptos/aptos.go
+++ b/data_layer/aptos/aptos.go
@@ -68,7 +68,8 @@ func NewAptos(root string) *Aptos {
 
 		daData, exists := consts.DaNetworks[aptConfig.Network]
 		if !exists {
-			panic(fmt.Errorf("DA network configuration not found for: %s", aptConfig.Network))
+			pterm.Error.Printf("DA network configuration not found for: %s", aptConfig.Network)
+			return &aptConfig
 		}
 
 		aptConfig.RpcEndpoint = daData.RpcUrl
@@ -105,11 +106,12 @@ func NewAptos(root string) *Aptos {
 		for {
 			proceed, _ := pterm.DefaultInteractiveConfirm.WithDefaultValue(false).
 				WithDefaultText(
-					"press 'y' when the wallet are funded",
+					"press 'y' when the wallet is funded",
 				).Show()
 
 			if !proceed {
-				panic(fmt.Errorf("APT addr need to be fund!"))
+				pterm.Error.Println("APT addr needs to be funded!")
+				continue
 			}
 
 			balance, err := aptConfig.getBalance()
@@ -123,7 +125,7 @@ func NewAptos(root string) *Aptos {
 				break
 			}
 
-			pterm.Println("APT wallet need to be fund!")
+			pterm.Error.Println("APT wallet needs to be funded!")
 		}
 
 		pterm.Warning.Print("You will need to save Private Key to an environment variable named APT_PRIVATE_KEY")

--- a/data_layer/avail/avail.go
+++ b/data_layer/avail/avail.go
@@ -80,12 +80,12 @@ func NewAvail(root string) *Avail {
 		} else {
 			entropySeed, err := bip39.NewEntropy(mnemonicEntropySize)
 			if err != nil {
-				panic(err)
+				pterm.Error.Println("failed to create new entropy", err)
 			}
 
 			availConfig.Mnemonic, err = bip39.NewMnemonic(entropySeed)
 			if err != nil {
-				panic(err)
+				pterm.Error.Println("failed to create new mnemonic", err)
 			}
 
 			fmt.Printf("\t%s\n", availConfig.Mnemonic)
@@ -95,19 +95,24 @@ func NewAvail(root string) *Avail {
 
 		keyringPair, err := signature.KeyringPairFromSecret(availConfig.Mnemonic, keyringNetworkID)
 		if err != nil {
-			panic(err)
+			pterm.Error.Println("failed to create new keypair from mnemonic", err)
 		}
 
 		pterm.DefaultSection.WithIndentCharacter("🔔").Println("Please fund your Avail addresses below")
 		pterm.DefaultBasicText.Println(pterm.LightGreen(keyringPair.Address))
 
-		proceed, _ := pterm.DefaultInteractiveConfirm.WithDefaultValue(false).
-			WithDefaultText(
-				"press 'y' when the wallets are funded",
-			).Show()
+		for {
+			proceed, _ := pterm.DefaultInteractiveConfirm.WithDefaultValue(false).
+				WithDefaultText(
+					"press 'y' when the wallets are funded",
+				).Show()
 
-		if !proceed {
-			panic(fmt.Errorf("Avail addr needs to be funded!"))
+			if !proceed {
+				pterm.Error.Println("Avail addr needs to be funded!")
+				continue
+			}
+
+			break
 		}
 
 		insufficientBalances, err := availConfig.CheckDABalance()
@@ -117,7 +122,7 @@ func NewAvail(root string) *Avail {
 
 		err = keys.PrintInsufficientBalancesIfAny(insufficientBalances)
 		if err != nil {
-			pterm.Error.Println("failed to check insufficient balances: ", err)
+			pterm.Error.Println("failed to check insufficient balances", err)
 		}
 
 		availConfig.RpcEndpoint = daData.ApiUrl
@@ -126,7 +131,7 @@ func NewAvail(root string) *Avail {
 
 		availConfig.AppID, err = CreateAppID(rollerData.DA.ApiUrl, availConfig.Mnemonic, rollerData.RollappID)
 		if err != nil {
-			panic(err)
+			pterm.Error.Println("failed to create AppID", err)
 		}
 
 		err = writeConfigToTOML(cfgPath, availConfig)

--- a/data_layer/avail/avail.go
+++ b/data_layer/avail/avail.go
@@ -65,7 +65,8 @@ func NewAvail(root string) *Avail {
 
 		daData, exists := consts.DaNetworks[daNetwork]
 		if !exists {
-			panic(fmt.Errorf("DA network configuration not found for: %s", daNetwork))
+			pterm.Error.Printf("DA network configuration not found for: %s", daNetwork)
+			return &availConfig
 		}
 
 		useExistingAvailWallet, _ := pterm.DefaultInteractiveConfirm.WithDefaultText(
@@ -106,7 +107,7 @@ func NewAvail(root string) *Avail {
 			).Show()
 
 		if !proceed {
-			panic(fmt.Errorf("Avail addr need to be fund!"))
+			panic(fmt.Errorf("Avail addr needs to be funded!"))
 		}
 
 		insufficientBalances, err := availConfig.CheckDABalance()

--- a/data_layer/avail/avail.go
+++ b/data_layer/avail/avail.go
@@ -63,6 +63,11 @@ func NewAvail(root string) *Avail {
 			daNetwork = string(consts.AvailTestnet)
 		}
 
+		daData, exists := consts.DaNetworks[daNetwork]
+		if !exists {
+			panic(fmt.Errorf("DA network configuration not found for: %s", daNetwork))
+		}
+
 		useExistingAvailWallet, _ := pterm.DefaultInteractiveConfirm.WithDefaultText(
 			"would you like to import an existing Avail wallet?",
 		).Show()
@@ -114,10 +119,6 @@ func NewAvail(root string) *Avail {
 			pterm.Error.Println("failed to check insufficient balances: ", err)
 		}
 
-		daData, exists := consts.DaNetworks[daNetwork]
-		if !exists {
-			panic(fmt.Errorf("DA network configuration not found for: %s", daNetwork))
-		}
 		availConfig.RpcEndpoint = daData.ApiUrl
 		availConfig.AccAddress = keyringPair.Address
 		availConfig.Root = root

--- a/data_layer/bnb/bnb.go
+++ b/data_layer/bnb/bnb.go
@@ -59,7 +59,8 @@ func NewBnb(root string) *Bnb {
 
 		daData, exists := consts.DaNetworks[daNetwork]
 		if !exists {
-			panic(fmt.Errorf("DA network configuration not found for: %b", daNetwork))
+			pterm.Error.Printf("DA network configuration not found for: %s", daNetwork)
+			return &bnbConfig
 		}
 
 		bnbConfig.RpcEndpoint = daData.RpcUrl
@@ -110,11 +111,12 @@ func NewBnb(root string) *Bnb {
 		for {
 			proceed, _ := pterm.DefaultInteractiveConfirm.WithDefaultValue(false).
 				WithDefaultText(
-					"press 'y' when the wallet are funded",
+					"press 'y' when the wallet is funded",
 				).Show()
 
 			if !proceed {
-				panic(fmt.Errorf("BNB addr need to be fund!"))
+				pterm.Error.Println("BNB addr needs to be funded!")
+				continue
 			}
 
 			balance, err := bnbConfig.getBalance()
@@ -127,7 +129,7 @@ func NewBnb(root string) *Bnb {
 				pterm.Println("Wallet funded with balance:", balance)
 				break
 			}
-			pterm.Println("BNB wallet need to be fund!")
+			pterm.Error.Println("BNB wallet needs to be funded!")
 		}
 
 		err = writeConfigToTOML(cfgPath, bnbConfig)

--- a/data_layer/bnb/bnb.go
+++ b/data_layer/bnb/bnb.go
@@ -76,7 +76,7 @@ func NewBnb(root string) *Bnb {
 			).Show()
 			privateKey, err := crypto.HexToECDSA(bnbConfig.PrivateKey)
 			if err != nil {
-				panic(err)
+				pterm.Error.Println("failed to parse private key from hex", err)
 			}
 			publicKey := privateKey.Public().(*ecdsa.PublicKey)
 			address := crypto.PubkeyToAddress(*publicKey).Hex()
@@ -85,15 +85,11 @@ func NewBnb(root string) *Bnb {
 		} else {
 			privateKey, err := crypto.GenerateKey()
 			if err != nil {
-				panic(err)
+				pterm.Error.Println("failed to generate new private key", err)
 			}
 
 			privateKeyBytes := crypto.FromECDSA(privateKey)
 			privateKeyHex := hex.EncodeToString(privateKeyBytes)
-			if err != nil {
-				panic(err)
-			}
-
 			bnbConfig.PrivateKey = privateKeyHex
 
 			fmt.Printf("\t%s\n", bnbConfig.PrivateKey)

--- a/data_layer/bnb/bnb.go
+++ b/data_layer/bnb/bnb.go
@@ -62,11 +62,14 @@ func NewBnb(root string) *Bnb {
 			panic(fmt.Errorf("DA network configuration not found for: %b", daNetwork))
 		}
 
-		useExistingbnbWallet, _ := pterm.DefaultInteractiveConfirm.WithDefaultText(
+		bnbConfig.RpcEndpoint = daData.RpcUrl
+		bnbConfig.Root = root
+
+		useExistingBnbWallet, _ := pterm.DefaultInteractiveConfirm.WithDefaultText(
 			"would you like to import an existing Bnb wallet?",
 		).Show()
 
-		if useExistingbnbWallet {
+		if useExistingBnbWallet {
 			bnbConfig.PrivateKey, _ = pterm.DefaultInteractiveTextInput.WithDefaultText(
 				"> Enter your hex private key",
 			).Show()
@@ -104,24 +107,27 @@ func NewBnb(root string) *Bnb {
 		pterm.DefaultSection.WithIndentCharacter("🔔").Println("Please fund your bnb addresses below")
 		pterm.DefaultBasicText.Println(pterm.LightGreen(bnbConfig.Address))
 
-		proceed, _ := pterm.DefaultInteractiveConfirm.WithDefaultValue(false).
-			WithDefaultText(
-				"press 'y' when the wallets are funded",
-			).Show()
+		for {
+			proceed, _ := pterm.DefaultInteractiveConfirm.WithDefaultValue(false).
+				WithDefaultText(
+					"press 'y' when the wallet are funded",
+				).Show()
 
-		if !proceed {
-			panic(fmt.Errorf("Bnb addr need to be fund!"))
-		}
+			if !proceed {
+				panic(fmt.Errorf("BNB addr need to be fund!"))
+			}
 
-		bnbConfig.RpcEndpoint = daData.RpcUrl
-		bnbConfig.Root = root
-		balance, err := bnbConfig.getBalance()
-		if err != nil {
-			panic(err)
-		}
+			balance, err := bnbConfig.getBalance()
+			if err != nil {
+				pterm.Println("Error getting balance:", err)
+				continue
+			}
 
-		if balance.Cmp(big.NewInt(0)) <= 0 {
-			panic(fmt.Errorf("Bnb wallet need to be fund!"))
+			if balance.Cmp(big.NewInt(0)) > 0 {
+				pterm.Println("Wallet funded with balance:", balance)
+				break
+			}
+			pterm.Println("BNB wallet need to be fund!")
 		}
 
 		err = writeConfigToTOML(cfgPath, bnbConfig)

--- a/data_layer/celestia/celestia.go
+++ b/data_layer/celestia/celestia.go
@@ -363,16 +363,18 @@ func (c *Celestia) GetSequencerDAConfig(nt string) string {
 
 	if nt == consts.NodeType.Sequencer {
 		authToken, err = c.getAuthToken(consts.DaAuthTokenType.Admin, raCfg)
+		if err != nil {
+			pterm.Error.Println("failed to get auth token", err)
+		}
 	} else if nt == consts.NodeType.FullNode {
 		authToken, err = c.getAuthToken(consts.DaAuthTokenType.Read, raCfg)
+		if err != nil {
+			pterm.Error.Println("failed to get auth token", err)
+		}
 	} else {
-		// TODO: don't panic,return an err
 		err := errors.New("invalid node type")
-		panic(err)
-	}
-
-	if err != nil {
-		panic(err)
+		pterm.Error.Println("failed to get auth token", err)
+		return ""
 	}
 
 	var daGasPrices float64

--- a/data_layer/loadnetwork/loadnetwork.go
+++ b/data_layer/loadnetwork/loadnetwork.go
@@ -76,7 +76,8 @@ func NewLoadNetwork(root string) *LoadNetwork {
 
 		daData, exists := consts.DaNetworks[daNetwork]
 		if !exists {
-			panic(fmt.Errorf("DA network configuration not found for: %s", daNetwork))
+			pterm.Error.Printf("DA network configuration not found for: %s", daNetwork)
+			return &loadNetworkConfig
 		}
 
 		useExistingWallet, _ := pterm.DefaultInteractiveConfirm.WithDefaultText(
@@ -109,11 +110,12 @@ func NewLoadNetwork(root string) *LoadNetwork {
 		for {
 			proceed, _ := pterm.DefaultInteractiveConfirm.WithDefaultValue(false).
 				WithDefaultText(
-					"press 'y' when the wallet are funded",
+					"press 'y' when the wallet is funded",
 				).Show()
 
 			if !proceed {
-				panic(fmt.Errorf("LoadNetwork wallet need to be fund!"))
+				pterm.Error.Println("LoadNetwork wallet needs to be funded!")
+				continue
 			}
 
 			balance, err := GetBalance(daData.ApiUrl, loadNetworkConfig.PrivateKey)
@@ -133,7 +135,7 @@ func NewLoadNetwork(root string) *LoadNetwork {
 				break
 			}
 
-			pterm.Println("LoadNetwork wallet need to be fund!")
+			pterm.Error.Println("LoadNetwork wallet needs to be funded!")
 		}
 
 		loadNetworkConfig.RpcEndpoint = daData.ApiUrl

--- a/data_layer/loadnetwork/loadnetwork.go
+++ b/data_layer/loadnetwork/loadnetwork.go
@@ -91,15 +91,11 @@ func NewLoadNetwork(root string) *LoadNetwork {
 		} else {
 			privateKey, err := crypto.GenerateKey()
 			if err != nil {
-				panic(err)
+				pterm.Error.Println("failed to generate private key", err)
+				return &loadNetworkConfig
 			}
 
 			privateKeyBytes := crypto.FromECDSA(privateKey)
-			// privateKeyHex := hex.EncodeToString(privateKeyBytes)
-			// if err != nil {
-			// 	panic(err)
-			// }
-
 			loadNetworkConfig.PrivateKey = fmt.Sprintf("%x", string(privateKeyBytes))
 
 			fmt.Printf("\t%s\n", loadNetworkConfig.PrivateKey)
@@ -120,13 +116,13 @@ func NewLoadNetwork(root string) *LoadNetwork {
 
 			balance, err := GetBalance(daData.ApiUrl, loadNetworkConfig.PrivateKey)
 			if err != nil {
-				pterm.Println("Error getting balance:", err)
+				pterm.Error.Println("Error getting balance:", err)
 				continue
 			}
 
 			balanceFloat, err := strconv.ParseFloat(balance, 64)
 			if err != nil {
-				pterm.Println("Error parsing balance:", err)
+				pterm.Error.Println("Error parsing balance:", err)
 				continue
 			}
 

--- a/data_layer/loadnetwork/loadnetwork.go
+++ b/data_layer/loadnetwork/loadnetwork.go
@@ -74,39 +74,67 @@ func NewLoadNetwork(root string) *LoadNetwork {
 			daNetwork = string(consts.LoadNetworkTestnet)
 		}
 
-		loadNetworkConfig.PrivateKey, _ = pterm.DefaultInteractiveTextInput.WithDefaultText(
-			"> Enter your PrivateKey without 0x",
-		).Show()
-
-		proceed, _ := pterm.DefaultInteractiveConfirm.WithDefaultValue(false).
-			WithDefaultText(
-				"press 'y' when the wallet are funded",
-			).Show()
-
-		if !proceed {
-			panic(fmt.Errorf("LoadNetwork wallet need to be fund!"))
-		}
-
 		daData, exists := consts.DaNetworks[daNetwork]
 		if !exists {
 			panic(fmt.Errorf("DA network configuration not found for: %s", daNetwork))
 		}
 
-		balance, err := GetBalance(daData.ApiUrl, loadNetworkConfig.PrivateKey)
-		if err != nil {
-			panic(err)
+		useExistingWallet, _ := pterm.DefaultInteractiveConfirm.WithDefaultText(
+			"would you like to import an existing LoadNetwork wallet?",
+		).Show()
+
+		if useExistingWallet {
+			loadNetworkConfig.PrivateKey, _ = pterm.DefaultInteractiveTextInput.WithDefaultText(
+				"> Enter your PrivateKey without 0x",
+			).Show()
+		} else {
+			privateKey, err := crypto.GenerateKey()
+			if err != nil {
+				panic(err)
+			}
+
+			privateKeyBytes := crypto.FromECDSA(privateKey)
+			// privateKeyHex := hex.EncodeToString(privateKeyBytes)
+			// if err != nil {
+			// 	panic(err)
+			// }
+
+			loadNetworkConfig.PrivateKey = fmt.Sprintf("%x", string(privateKeyBytes))
+
+			fmt.Printf("\t%s\n", loadNetworkConfig.PrivateKey)
+			fmt.Println()
+			fmt.Println(pterm.LightYellow("💡 save this information and keep it safe"))
 		}
 
-		balanceFloat, err := strconv.ParseFloat(balance, 64)
-		if err != nil {
-			panic(err)
-		}
+		for {
+			proceed, _ := pterm.DefaultInteractiveConfirm.WithDefaultValue(false).
+				WithDefaultText(
+					"press 'y' when the wallet are funded",
+				).Show()
 
-		if balanceFloat == 0 {
-			panic(fmt.Errorf("LoadNetwork wallet need to be fund!"))
-		}
+			if !proceed {
+				panic(fmt.Errorf("LoadNetwork wallet need to be fund!"))
+			}
 
-		pterm.Println("LoadNetwork Balance: ", balanceFloat)
+			balance, err := GetBalance(daData.ApiUrl, loadNetworkConfig.PrivateKey)
+			if err != nil {
+				pterm.Println("Error getting balance:", err)
+				continue
+			}
+
+			balanceFloat, err := strconv.ParseFloat(balance, 64)
+			if err != nil {
+				pterm.Println("Error parsing balance:", err)
+				continue
+			}
+
+			if balanceFloat > 0 {
+				pterm.Println("Wallet funded with balance:", balance)
+				break
+			}
+
+			pterm.Println("LoadNetwork wallet need to be fund!")
+		}
 
 		loadNetworkConfig.RpcEndpoint = daData.ApiUrl
 		loadNetworkConfig.Root = root

--- a/data_layer/sui/sui.go
+++ b/data_layer/sui/sui.go
@@ -62,7 +62,8 @@ func NewSui(root string) *Sui {
 
 		daData, exists := consts.DaNetworks[daNetwork]
 		if !exists {
-			panic(fmt.Errorf("DA network configuration not found for: %s", daNetwork))
+			pterm.Error.Printf("DA network configuration not found for: %s", daNetwork)
+			return &suiConfig
 		}
 
 		useExistingSuiWallet, _ := pterm.DefaultInteractiveConfirm.WithDefaultText(
@@ -103,7 +104,7 @@ func NewSui(root string) *Sui {
 			).Show()
 
 		if !proceed {
-			panic(fmt.Errorf("Sui addr need to be fund!"))
+			panic(fmt.Errorf("Sui addr needs to be funded!"))
 		}
 
 		suiConfig.RpcEndpoint = daData.RpcUrl

--- a/data_layer/sui/sui.go
+++ b/data_layer/sui/sui.go
@@ -77,12 +77,14 @@ func NewSui(root string) *Sui {
 		} else {
 			entropySeed, err := bip39.NewEntropy(MnemonicEntropySize)
 			if err != nil {
-				panic(err)
+				pterm.Error.Println("failed to generate new entropy:", err)
+				return &suiConfig
 			}
 
 			suiConfig.Mnemonic, err = bip39.NewMnemonic(entropySeed)
 			if err != nil {
-				panic(err)
+				pterm.Error.Println("failed to generate new mnemonic:", err)
+				return &suiConfig
 			}
 
 			fmt.Printf("\t%s\n", suiConfig.Mnemonic)
@@ -92,19 +94,24 @@ func NewSui(root string) *Sui {
 
 		key, err := signer.NewSignertWithMnemonic(suiConfig.Mnemonic)
 		if err != nil {
-			panic(err)
+			pterm.Error.Println("failed to generate new signer from mnemonic:", err)
+			return &suiConfig
 		}
 
 		pterm.DefaultSection.WithIndentCharacter("🔔").Println("Please fund your sui addresses below")
 		pterm.DefaultBasicText.Println(pterm.LightGreen(key.Address))
 
-		proceed, _ := pterm.DefaultInteractiveConfirm.WithDefaultValue(false).
-			WithDefaultText(
-				"press 'y' when the wallets are funded",
-			).Show()
+		for {
+			proceed, _ := pterm.DefaultInteractiveConfirm.WithDefaultValue(false).
+				WithDefaultText(
+					"press 'y' when the wallets are funded",
+				).Show()
 
-		if !proceed {
-			panic(fmt.Errorf("Sui addr needs to be funded!"))
+			if !proceed {
+				pterm.Error.Println("Sui addr needs to be funded!")
+				continue
+			}
+			break
 		}
 
 		suiConfig.RpcEndpoint = daData.RpcUrl

